AreaShop/src/main/java/me/wiefferink/areashop/ASLanguageManager.java

```diff
@@ -0,0 +1,175 @@
+        this.plugin = plugin;
+        this.jarLanguagePath = jarLanguagePath;
+        this.chatPrefix = chatPrefix;
+        this.languageFolder = new File(plugin.getDataFolder() + File.separator + jarLanguagePath);
+
+        Message.init(this, plugin.getLogger());
+        saveDefaults();
+        currentLanguage = loadLanguage(currentLanguageName);
+        if(defaultLanguageName.equals(currentLanguageName)) {
+            defaultLanguage = currentLanguage;
+        } else {
+            defaultLanguage = loadLanguage(defaultLanguageName);
+        }
+        this.wgPrefix = wgPrefix;
+    }
+
+    @Override
+    public List<String> getMessage(String key) {
+        List<String> message;
+        if(key.equalsIgnoreCase(Message.CHATLANGUAGEVARIABLE)) {
+            message = chatPrefix;
+        } else if(key.equalsIgnoreCase("wgPrefix"))
+        {
+            message = new ArrayList<>();
+            message.add(wgPrefix);
+        } else if(currentLanguage.containsKey(key)) {
+            message = currentLanguage.get(key);
+        } else {
+            message = defaultLanguage.get(key);
+        }
+        if(message == null) {
+            Log.warn("Did not find message '" + key + "' in the current or default language");
+            return new ArrayList<>();
+        }
+        return new ArrayList<>(message);
+    }
+
+    private void saveDefaults() {
+        // Create the language folder if it not exists
+        File langFolder;
+        if(!languageFolder.exists()) {
+            if(!languageFolder.mkdirs()) {
+                Log.warn("Could not create language directory: " + languageFolder.getAbsolutePath());
+                return;
+            }
+        }
+
+        try {
+            // Read jar as ZIP file
+            File jarPath = new File(plugin.getClass().getProtectionDomain().getCodeSource().getLocation().toURI());
+            ZipFile jar = new ZipFile(jarPath);
+            Enumeration<? extends ZipEntry> entries = jar.entries();
+
+            // Each entry is a file or directory
+            while(entries.hasMoreElements()) {
+                ZipEntry entry = entries.nextElement();
+
+                // Filter to YAML files in the language directory
+                if(!entry.isDirectory() && entry.getName().startsWith(jarLanguagePath+"/") && entry.getName().endsWith(".yml")) {
+                    // Save the file to disk
+                    File targetFile = new File(languageFolder.getAbsolutePath() + File.separator + entry.getName().substring(entry.getName().lastIndexOf("/")));
+                    try(
+                            InputStream input = jar.getInputStream(entry);
+                            OutputStream output = new FileOutputStream(targetFile)
+                    ) {
+                        int read;
+                        byte[] bytes = new byte[1024];
+                        while((read = input.read(bytes)) != -1) {
```

Comment:
I think you can use `InputStream#readAllBytes`

Reply:
Resolved